src/workspaceConfig.ts

```diff
@@ -45,16 +44,27 @@ export function updateIdfComponentsTree(projectDescriptionPath: string) {
     idfDataProvider.refresh(projectDescriptionPath);
 }
 
-export function updateProjectName(projectDescriptionPath: string) {
-    if (!utils.fileExists(projectDescriptionPath)) {
-        return;
-    }
-    fs.readFile(projectDescriptionPath, (err, data) => {
-        if (err) {
-            Logger.errorNotify(err.message, err);
-            return;
+export function getProjectName(workspace: string): Promise<string> {
+    const cmakeListFile = path.join(workspace, "CMakeLists.txt");
+    return new Promise((resolve, reject) => {
+        try {
+            if (!utils.fileExists(cmakeListFile)) {
+                reject(`${cmakeListFile} doesn't exist.`);
+            }
+            fs.readFile(cmakeListFile, (err, data) => {
+                if (err) {
+                    Logger.errorNotify(err.message, err);
```

Comment:
maybe use `Logger.error()` instead of `errorNotify` since this is an utility function, and caller should decide how to display the error